### PR TITLE
Add simple boss health info frame

### PR DESCRIPTION
**Pull request by emmericp**
Displays the boss health that DBM keeps track of anyways. Can only be used for creature IDs that are explicitly registered for the mod

### CODE REVIEW

File: DBM-Core/DBM-Core.lua [hunk 1]
```diff
@@ -8584,9 +8587,27 @@ function DBM:GetBossHP(cIdOrGUID, onlyHighest)
 					bossHealth[cIdOrGUID] = hp
 				end
 				bossHealthuIdCache[cIdOrGUID] = unitId
+				bossNames[cIdOrGUID] = UnitName(unitId)
 				return hp, unitId, UnitName(unitId)
 			end
 		end
+		if not isRetail then
```
Comment (MysticalOS) [inline DBM-Core/DBM-Core.lua:8594]: Wouldn't it make sense to replace this entire classic block with just GetUnitIdFromCID or GetUnitIdFromGUID which both have automatic iteration selection for retail/classic, and will even use blizzards UnitTokenFromGUID function if available (guid checks only)

Based on your code i think you'd probably use GetUnitIdFromCID though which can be DBM or bossmodprototype/mod (I'll touch GetUnitIdFromCID up a little to not scan boss unit ids on classic era. although it doesn't error, it just wastes time)

**Review by MysticalOS — CHANGES_REQUESTED** [review]
Relying only nameplates is prone to failure with 20 yard range and possibility player might have them off. using player target iteration has always been more robust. which is handled automatically by GetUnitIdFromCID (and if needed nameplate iterator can be added to that function as well for classic era)

**Comment by MysticalOS** [comment]
https://github.com/DeadlyBossMods/DBM-Unified/commit/09725639a67c2e4abcacc507645f1070c8ecc10c just modernized GetUnitIdFromCID a bit so it's far more robust now as a utility function, matching it's counterpart GetUnitIdFromGUID

File: DBM-Core/DBM-Core.lua (hunk 1 above)
Comment (emmericp) [reply DBM-Core/DBM-Core.lua:8594]: no, because i want this to happen as a fallback after scanning raid/party targets failed, not before

**Comment by emmericp** [comment]
i think you misread this: it's not replacing the existing mechanism, it adds a new fallback option if party/raid target scanning doesn't find it.

the situation where this is potentially useful is Menagerie fight when reflect is up on the sheep but dots are still ticking

**Comment by MysticalOS** [comment]
Ah, well fallback no longer needed either, cause i updated method to scan same table GUID scan does which includes nameplates

**Comment by emmericp** [comment]
i agree that the code duplication is no good, but the new method has a few problems which is why i would prefer this over the new one for the time being:

1) no caching
2) unconditionally scans every possible target independent of party/raid or size

**Review by MysticalOS — APPROVED** [review]

**Pull request merged**